File: Euclid/Util/DateUtil.cs
using System;

namespace Euclid.Util
{
    public class DateUtil
    {
        /// <summary>
        /// Get the current unix timestamp.
        /// </summary>
        public static long GetUnixTimestamp()
        {
            return DateTimeOffset.Now.ToUnixTimeSeconds();
        }

        /// <summary>
        /// Get the current unix timestamp.
        /// </summary>
        public static long GetUnixTimestampInMillis()
        {
            return DateTimeOffset.Now.ToUnixTimeMilliseconds();
        }
    }
}

File: euclid/src/main/java/euclid/util/DateUtil.java
package euclid.util;

import java.time.Instant;

public class DateUtil {

    public static long getUnixTimestamp() {
        return Instant.now().getEpochSecond();
    }

    public static long getUnixTimestampInMillis() {
        return Instant.now().toEpochMilli();
    }
}
